R/parseTimeseries.R

```diff
@@ -65,21 +77,25 @@ parseTimeseries <- function(file, delim, with.units = FALSE){
   return(dataOut)
 }
 
-parseConfig = function(file, delim){
-  featureLine = 2
-  skipHead = 1
-  varMarker = '# '
-  c <- file(file,"r") 
-  fileLines <- readLines(c)
-  close(c)
-  nRead <- length(fileLines)
-  blockStart <- grep(varMarker, fileLines)
-  skips = blockStart+skipHead
-  blockEnd = c(blockStart[-1] - 1, nRead)
-  nrows = blockEnd - skips - 1
-  features = unique(strsplit(fileLines[featureLine], split = delim)[[1]][-1])
-  vars = sub(varMarker,"",fileLines[blockStart])
-  config = list(vars = vars, features = features, skip = skips, nrows = nrows)
+parseConfig = function(file, delim, largeFile=FALSE){
+  if(largeFile==FALSE){
+    featureLine = 2 # Line containing unique IDs of features (stencil) that were processed
+    skipHead = 1 # Number of lines to skip past the variable marker header?
+    varMarker = '# ' # Symbol that denotes a variable identifier and a new block of output.
+    c <- file(file,"r") 
+    fileLines <- readLines(c) # This concerns me!!!
```

Comment:
We should discuss this, I have some situations (national HUC12s against daily 30 year data for example) where the GDP output file may be big enough to fit in a data frame but opening the whole ascii file might exceed system resources.